extras/src/com/mopub/mobileads/AdColonyRewardedVideo.java

```
@@ -148,24 +154,65 @@ protected void loadWithSdkInitialized(@NonNull final Activity activity,
             mAdUnitId = (String) adUnitObject;
         }
 
-        mAd = new AdColonyV4VCAd(mZoneId).withListener(sAdColonyListener);
-        sAdToZoneIdMap.put(mAd, mZoneId);
+        sZoneIdToAdMap.put(mZoneId, null);
+        setUpAdOptions();
+        mAdColonyListener = new AdColonyListener(mAdColonyAdOptions);
+        AdColony.setRewardListener(mAdColonyListener);
+        AdColony.requestInterstitial(mZoneId, mAdColonyListener, mAdColonyAdOptions);
         scheduleOnVideoReady();
     }
 
+    private void setUpAdOptions() {
+        mAdColonyAdOptions.enableConfirmationDialog(getConfirmationDialogFromSettings());
+        mAdColonyAdOptions.enableResultsDialog(getResultsDialogFromSettings());
+    }
+
+    private void setAppOptions(String clientOptions) {
+        if (clientOptions == null || clientOptions.isEmpty()) {
+            Log.d(TAG, "AdColony client options are not configured on the MoPub dashboard");
+            return;
+        }
+
+        String[] allOptions = clientOptions.split(",");
+        for (String option : allOptions) {
+            String optionNameAndValue[] = option.split(":");
+            if (optionNameAndValue.length == 2) {
+                switch (optionNameAndValue[0]) {
+                    case "store":
+                        mAdColonyAppOptions.setOriginStore(optionNameAndValue[1]);
+                        break;
+                    case "version":
+                        mAdColonyAppOptions.setAppVersion(optionNameAndValue[1]);
```

instead of returning void, this method prob should return a boolean to indicate success or failure